### PR TITLE
PRIME-1055 Address autocomplete bug omits province 

### PR DESCRIPTION
**Pull request by mtpultz**

### CODE REVIEW

File: prime-angular-frontend/src/app/shared/components/address-autocomplete/address-autocomplete.component.html [hunk 1]
```diff
@@ -8,14 +8,12 @@
         <input matInput
                placeholder="Address Autocomplete"
                formControlName="autocomplete"
-               required="false"
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/shared/components/address-autocomplete/address-autocomplete.component.html:11]: @AnaisHebert controllers should be used to drive reactive forms, and this should be used only if not attached to a form in the controller

File: prime-angular-frontend/src/app/shared/components/address-autocomplete/address-autocomplete.component.ts [hunk 2]
```diff
@@ -32,40 +36,44 @@ export class AddressAutocompleteComponent implements OnInit {
   }
 
   public onAutocomplete(id: string) {
-    this.addressRetrieved = null;
     this.addressAutocompleteResource.retrieve(id)
-      .subscribe((response: AddressAutocompleteRetrieveResponse[]) => {
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/shared/components/address-autocomplete/address-autocomplete.component.ts:37]: @AnaisHebert did a bit of clean up in this method and dropped the use of a closure

File: prime-angular-frontend/src/app/shared/components/address-autocomplete/address-autocomplete.component.ts [hunk 3]
```diff
@@ -32,40 +36,44 @@ export class AddressAutocompleteComponent implements OnInit {
...
 
   public ngOnInit(): void {
-    this.form = new FormGroup({ autocomplete: new FormControl() });
+    this.form = this.fb.group({ autocomplete: [''] });
 
     this.autocomplete.valueChanges
-      .subscribe(() => {
-        this.addressAutocompleteFields = [];
-        if (this.autocomplete.value) {
-          this.addressAutocompleteResource.find(this.autocomplete.value)
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/shared/components/address-autocomplete/address-autocomplete.component.ts:64]: @AnaisHebert observables don't like to be in observables.  This is an anti-pattern that can result in some issues, but will most likely work 90% of the time.  You want to switch what you're observing using operators like `exhaustMap`, and then block if not further processing is required using `EMPTY`

File: prime-angular-frontend/src/app/shared/components/forms/address/address.component.ts [hunk 4]
```diff
@@ -75,8 +75,11 @@ export class AddressComponent implements OnInit {
     return this.formUtilsService.isRequired(this.form, path);
   }
 
-  public onAutocomplete({ id, ...remainder }: Address) {
-    this.form.patchValue(remainder);
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/shared/components/forms/address/address.component.ts:79]: The culprit!

File: prime-angular-frontend/src/app/shared/components/address-autocomplete/address-autocomplete.component.ts [hunk 5]
```diff
@@ -32,40 +36,44 @@ export class AddressAutocompleteComponent implements OnInit {
...
-          ? this.autocompleteAddress.emit(address)
-          : this.toastService.openErrorToast('Address must be in BC');
+        } else {
+          this.toastService.openErrorToast('Address could not be retrieved');
+        }
       });
   }
 
   public ngOnInit(): void {
-    this.form = new FormGroup({ autocomplete: new FormControl() });
```
Comment (mtpultz) [inline prime-angular-frontend/src/app/shared/components/address-autocomplete/address-autocomplete.component.ts:58]: @AnaisHebert we'll want to use the `FormBuilder` helper for form creation

File: prime-angular-frontend/src/app/shared/components/forms/address/address.component.ts (hunk 4 above)
Comment (AnaisHebert) [reply prime-angular-frontend/src/app/shared/components/forms/address/address.component.ts:79]: Yayyyyy!

**Review by AnaisHebert — APPROVED** [review]
🍕🍕🍕🍟🍟🍔

**Pull request merged**